crates/router/src/db/payment_method.rs

```diff
@@ -61,6 +72,297 @@ pub trait PaymentMethodInterface {
+                    .apply_changeset(payment_method.clone());
+                
+                    let redis_value = serde_json::to_string(&updated_payment_method)
+                        .into_report()
+                        .change_context(errors::StorageError::SerializationFailed)?;
+
+                let redis_entry = kv::TypedSql {
+                    op: kv::DBOperation::Update {
+                        updatable: kv::Updateable::PaymentMethodUpdate(kv::PaymentMethodUpdateMems {
+                            orig: payment_method,
+                            update_data: p_update,
+                        }),
+                    },
+                };
+
+                kv_wrapper::<(), _, _>(
+                    self,
+                    KvOperation::<diesel_models::PaymentMethod>::Hset((&field, redis_value), redis_entry),
+                    &key,
+                )
+                .await
+                .map_err(|err| err.to_redis_failed_response(&key))?
+                .try_into_hset()
+                .change_context(errors::StorageError::KVError)?;
+
+                Ok(updated_payment_method)
+            }
+        }
+    }
+
+    #[instrument(skip_all)]
+    async fn find_payment_method_by_customer_id_merchant_id_list(
+        &self,
+        customer_id: &str,
+        merchant_id: &str,
+        limit: Option<i64>,
+    ) -> CustomResult<Vec<storage::PaymentMethod>, errors::StorageError> {
+        let conn = connection::pg_connection_read(self).await?;
+        storage::PaymentMethod::find_by_customer_id_merchant_id(
+            &conn,
+            customer_id,
+            merchant_id,
+            limit,
+        )
+        .await
+        .map_err(|error| report!(errors::StorageError::from(error)))
+    }
+
+    #[instrument(skip_all)]
+    async fn find_payment_method_by_customer_id_merchant_id_status(
+        &self,
+        customer_id: &str,
+        merchant_id: &str,
+        status: common_enums::PaymentMethodStatus,
+        limit: Option<i64>,
+    ) -> CustomResult<Vec<storage::PaymentMethod>, errors::StorageError> {
+        let conn = connection::pg_connection_read(self).await?;
+        storage::PaymentMethod::find_by_customer_id_merchant_id_status(
+            &conn,
+            customer_id,
+            merchant_id,
+            status,
+            limit,
+        )
+        .await
+        .map_err(|error| report!(errors::StorageError::from(error)))
+    }
+
+    async fn delete_payment_method_by_merchant_id_payment_method_id(
+        &self,
+        merchant_id: &str,
+        payment_method_id: &str,
+    ) -> CustomResult<storage::PaymentMethod, errors::StorageError> {
+        let conn = connection::pg_connection_write(self).await?;
+        storage::PaymentMethod::delete_by_merchant_id_payment_method_id(
+            &conn,
+            merchant_id,
+            payment_method_id,
+        )
+        .await
+        .map_err(|error| report!(errors::StorageError::from(error)))
+    }
```

Comment:
Let's say when I do an insert_payment_method and data is not inserted into the drainer and immediately I do a delete this will cause error and data will still be in redis.

Reply:
yes,  we have to start supporting delete in kv as well (this is the same case as find all as well right where we are eventually consistent)